cruise-control-client/cruisecontrolclient/client/Responder.py

```diff
@@ -16,21 +22,104 @@
 # To be able to define a multithreaded way of interacting with cruise-control
 from threading import Thread
 
+# To be able to deprecate classes
+import warnings
+
+
+class CruiseControlResponder(requests.Session):
+    """
+    This class is intented to lightly wrap requests' Session class,
+    in order to provide the cruise-control-client with some basic
+    sanity checking and session-management functionality.
+    """
+
+    def send(self, request, **kwargs):
+        # Alert the humans to long-running poll
+        print_error(f"Starting long-running {request.method} of {request.url}")
+        if request.body:
+            print_error(f"body:{request.body.decode()}")
+
+        # Convenience closure to not have to copy-paste the parameters from
+        # this current environment.
+        def inner_send_helper():
+            return super(CruiseControlResponder, self).send(request, **kwargs)
+
+        # Loop our requests until we get a "final" response
+        response = inner_send_helper()
+        while 'progress' in response.json().keys():
+            display_response(response)
+            response = inner_send_helper()
+
+        return response
+
+    def retrieve_response(self, method, url, **kwargs) -> requests.Response:
+        """
+        Returns a final requests.Response object from cruise-control
+        where Response.text is JSON-formatted.
+
+        :return: requests.Response
+        """
+        # cruise-control's JSON response has a 'progress' key in it so long
```

Comment:
Just FYI for the upcoming feature: https://github.com/linkedin/cruise-control/pull/876